mindsdb/integrations/handlers/sharepoint_handler/sharepoint_api.py

```diff
@@ -0,0 +1,429 @@
+
+    def disconnect(self) -> None:
+        self.bearer_token = None
+        self.is_connected = False
+
+    def get_all_sites(self, limit: int = None) -> List[Dict[Text, Any]]:
+        url = "https://graph.microsoft.com/v1.0/sites?search=*"
+
+        payload = {}
+        headers = {
+            "Authorization": f"Bearer {self.bearer_token}",
+            "Content-Type": "application/json",
+        }
+
+        response = self.getresponse(
+            request_type="GET", url=url, headers=headers, payload=payload, files=[]
+        )
+        response = response.json()["value"]
+        if limit:
+            response = response[:limit]
+
+        return response
+
+    def update_sites(
+        self, site_dict: List[Dict[Text, Text]], values_to_update: Dict[Text, Any]
+    ) -> None:
+        for site_entry in site_dict:
+            self.update_a_site(
+                site_id=site_entry["siteId"],
+                values_to_update=values_to_update,
+            )
+
+    def update_a_site(self, site_id: str, values_to_update: Dict[Text, Any]) -> None:
+        url = f"https://graph.microsoft.com/v1.0/sites/{site_id}/"
+        payload = values_to_update
+        payload = json.dumps(payload, indent=2)
+        headers = {
+            "Authorization": f"Bearer {self.bearer_token}",
+            "Content-Type": "application/json",
+        }
+        self.getresponse(
+            request_type="PATCH", url=url, headers=headers, payload=payload, files=[]
+        )
+
+    def get_lists_by_site(
+        self, site_id: str, limit: int = None
+    ) -> List[Dict[Text, Any]]:
+        url = f"https://graph.microsoft.com/v1.0/sites/{site_id}/lists"
+        payload = {}
+        headers = {
+            "Authorization": f"Bearer {self.bearer_token}",
+            "Content-Type": "application/json",
+        }
+
+        response = self.getresponse(
+            request_type="GET", url=url, headers=headers, payload=payload, files=[]
+        )
+        response = response.json()["value"]
+        if limit:
+            response = response[:limit]
+        return response
+
+    def get_all_lists(self, limit: int = None) -> List[Dict[Text, Any]]:
+        sites = self.get_all_sites()
+        lists = []
+        for site in sites:
+            for list_dict in self.get_lists_by_site(site_id=site["id"].split(",")[1]):
+                list_dict["siteName"] = site["name"]
+                list_dict["siteId"] = site["id"].split(",")[1]
+                lists.append(list_dict)
+        if limit:
+            lists = lists[:limit]
+        return lists
+
+    def delete_lists(self, list_dict: List[Dict[Text, Any]]) -> None:
+        for list_entry in list_dict:
+            self.delete_a_list(site_id=list_entry["siteId"], list_id=list_entry["id"])
+
+    def delete_a_list(self, site_id: str, list_id: str) -> None:
+        url = f"https://graph.microsoft.com/v1.0/sites/{site_id}/lists/{list_id}"
+        payload = {}
+        headers = {
```

Comment:
We can move this to utility function as most of the methods bellow are using it

Reply:
resolved in d4be410003d92952b9893e82ab59261bb6499a4a